File: src/selectionSort/selectionSort.py
 #!/usr/bin/python
# -*- coding: utf-8 -*-

###################################################
#                       					      
# A simple sorting program using Insertion Sort   
# and vectors in Python 					      
# Author: Jesús Iván Gastelum Romero              
# Contributors: Luis Edmundo Espinoza Larios     
#                       					      
###################################################

from time import time

def selectionSort( v ):
	j=0
	N = len(v)
	for i in range(0, N-1):
		pos_min = i
		j = i + 1

		while j  < N:
			if(v[j] < v[pos_min]):
				pos_min = j
			j = j + 1
		tmp = v[i]
		v[i] = v[pos_min]
		v[pos_min] = tmp
	print v

#*-----------------------------*#
#         MAIN PROGRAM          #
#*-----------------------------*#

start_time = time()
VECTOR = [7 , -5 , 0, 2 , 16, -9, 21, -874, 0, 8, 16, -210]
selectionSort(VECTOR)
end_time = time() - start_time
print '\nExecution Time: ', end_time, '\n'

File: src/selectionSort/selectionSort.c
/*
 * An implementation of the selection sort algorithm
 * Author Luis Edmundo Espinoza Larios
 * Contributors Jesús Iván Gastelum Romero
 */

#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <sys/time.h>
#include "util.h"

/*
 * Sorts the given array using the selection sort algorithm
 */
void sort(int[]);

int main()
{
    static int array[SIZE];
    struct timeval start, end;

    srand(time(NULL));

    generateArray(array);

    gettimeofday(&start, NULL);

    sort(array);

    gettimeofday(&end, NULL);

    printArray(array);

    printf ("Execution time: %f us\n", (double) (end.tv_usec - start.tv_usec) / 1000000 + (double) (end.tv_sec - start.tv_sec));

    return 0;
}

void sort(int array[])
{
    int i, j, lowestNumberIndex;

    for (i = 0; i < SIZE - 1; i++) {
        lowestNumberIndex = i;

        for (j = i + 1; j < SIZE; j++) {

            if (array[j] < array[lowestNumberIndex]) {
                lowestNumberIndex = j;
            }
        }

        swap(&array[i], &array[lowestNumberIndex]);
    }
}
